### PR TITLE
Fix/ab#83183 default value not working with ref data

### PR DESCRIPTION
**Pull request by NathanHGit**
# Description

When using reference data for a question, the default value popup was not working correctly. The data was being stored in `.visibleChoices` instead of `.choices` to prevent the loaded data from appearing in the choices tab. Unfortunately, the default value pop up is based on `.choices` and as a result, nothing was displayed. It is a surveyjs feature and it cannot be replaced.

While there are no issues if reference data choices appear in the choices tab, it may not be very intuitive. We looked for a solution with Matthis. Here are the results:
- Allow only one "choices" tab expansion at a time
- By default, the current one will be opened and won't be closable
- When clicking on other "choices" tab, previous values will be cleared
- A confirmation pop-up will inform the user of this behavior

SurveyCreator and PanelModel don't have an event that listens for expanding or collapsing tabs, so I used `PanelModel.onPropertyChanged`. Since I couldn't listen before expanding a tab, I had to collapse it again. This approach may seem less than ideal, but I haven't found a better solution.

I moved the `new SurveyModel` of the filter-builder-modal above the `renderGlobalProperties()` because reference data weren't loaded.

## Useful links

- [Please insert link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/83183/)
- [PanelModel.onPropertyChanged](https://surveyjs.io/form-library/documentation/api-reference/panel-model#onPropertyChanged)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Bug fix: I created a form with questions linked to reference data and checked if the default value modal was loading correctly.
New feature: I created a form with a dropdown question and switched between choices tabs to control the behavior.

## Screenshots

Bug fix

peek

New feature

peek2

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

**Comment by AntoineRelief** [comment]
@NathanHGit @Matthis-M-ReliefApps 
I think we could have a look at how questions behave when using choices from restful service, because for this kind of configuration I don't think we use the "choices" field, but the default value can be set

File: libs/shared/src/lib/survey/global-properties/reference-data.ts [hunk 1]
```diff
@@ -242,10 +242,8 @@ export const render = (
             filter
           )
           .then((choices) => {
-            question.choices = [];
-            // this is to avoid that the choices appear on the 'choices' tab
             question.setPropertyValue(
-              'visibleChoices',
+              'choices',
```
Comment (AntoineRelief) [inline libs/shared/src/lib/survey/global-properties/reference-data.ts]: @NathanHGit 

does this line save the choices in the DB?

cc @Matthis-M-ReliefApps

Comment (NathanHGit) [reply libs/shared/src/lib/survey/global-properties/reference-data.ts]: Yes it does, I will add code to empty the choices before saving the form.

**Comment by AntoineRelief** [comment]
@NathanHGit 
another comment
where did you find the default popup would use only "choices"?

**Comment by NathanHGit** [comment]
@AntoineRelief I didn't really find it, it's more something I deduced. I searched through the doc and various forums, but it doesn't seem possible. I didn't express myself well in the pull request, sorry

**Comment by AntoineRelief** [comment]
there may be something to do with 
```
this.surveyCreator.onPropertyGridShowModal.add(
      (sender: any, options: any) => {
        // and there, put some code
      }
    );
```

I tested it quickly, it seems it can be quite similar to what we have done there, for example in the form-builder.component file:

```
    this.surveyCreator.survey.onAfterRenderQuestion.add(
      renderGlobalProperties(this.referenceDataService) as any
    );
    this.surveyCreator.survey.onAfterRenderQuestion.add(
      this.formHelpersService.addQuestionTooltips as any
    );
```

I still don't understand exactly to which element the similar logic should be applied, but my guess is that the custom properties we build ( tooltip / ref data ) are not available in the modal

**Comment by NathanHGit** [comment]
The last push isn't an improvement but tests.

As pointed out before, using the "choices" property causes the data to be saved in the database. Data is duplicated which is not a viable solution. To prevent it we can clear "choices" before saving the survey. I haven't implemented it.

As proposed by Antoine, I tried to use [onPropertyGridShowModal](https://surveyjs.io/survey-creator/documentation/api-reference/survey-creator#onPropertyGridShowModal). I couldn't make it work but it seems possible. 

The objects given by the function are spreaded out, and I couldn't find the correct property to replace. At best, it only worked after reopening the popup. According to the documentation (which I double checked), the event is called before the pop up loads. However, in my case, the replaced data wasn't taken into account directly. I concluded that I was dealing with the wrong property.

**Comment by AntoineRelief** [comment]
closing this one as it does not solve

**Pull request closed without merging**